pkg/sampling/encoding_test.go

```diff
@@ -0,0 +1,376 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package sampling
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"math/rand"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+)
+
+func must[T any](t T, err error) T {
+	if err != nil {
+		panic(err)
+	}
+	return t
+}
+
+func mustNot[T any](_ T, err error) error {
+	if err == nil {
+		return fmt.Errorf("expected an error, got nil")
```

Comment:
use `require.Error` here

Reply:
I added comments to explain why these are here and why the test is more readable as a result.